### DISCUSSION TITLE
make compose_build stuck at `RUN yarn build`

### DISCUSSION THREAD

**Discussion by eugeneihde** [getredash/redash#6896] [Q&A] [answered]
I did everything for the setup that was written in the documentation.
Then i ran the command `make compose_build`. Now the build process is like stuck at ` => [server frontend-builder-2 11/11] RUN yarn build`. 
It always repeats the same output like this for the last 30 minutes:
`=> => #  8: 0xf36d20 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/usr/local/bin/node]`
`=> => #  9: 0xf130c8 v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/usr/local/bin/node]`
`=> => # 10: 0xf140a0 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/usr/local/bin/node]`
`=> => # 11: 0xef6d30 v8::internal::Factory::NewFillerObject(int, v8::internal::AllocationAlignment, v8::internal::AllocationType, v8::internal::AllocationOrigin) [/usr/local/bin/node]`
`=> => # 12: 0x129d42c v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [/usr/local/bin/node]`
`=> => # 13: 0x1681fac  [/usr/local/bin/node] `

**Comment by justinclift** [comment]
Oh, that's weird.  I ran through that earlier today and it didn't error out.

What OS are you running on, which version of Node are you using, and what are the specs of your system (cpus, ram)?

**Reply by justinclift** [reply]
For reference, I've been manually going through the steps today and yesterday in an Ubuntu 20.04 VM with 8GB of ram and between 6 and 16 cpus (varies).

The 8GB of ram is probably somewhat important, as Docker seems to eat ram for breakfast.  The number of cpus is probably less important, as it'll just take longer to finish rather than crashing.  Hopefully. :smile:

**Reply by eugeneihde** [reply]
I dont think it throws an Error. It just somehow always repeats the Output from the `RUN yarn build` Task.
I am using MacOS Sonoma 14.4.1 with Apple M1 and 16GB Ram

**Reply by justinclift** [reply]
Ahhh.  It could just be taking a super long time to compile.

When I compile it with a 6 core VM, and forget to set cpu affinity, then the compile process can take like 35+ minutes.

Since you're running on macOS, it's probably running Docker Desktop which (from rough memory) still does everything inside a Linux VM.  Maybe go have food/lunch/something and see if an extra 1/2 hour gives a result? :smile:

**Reply by justinclift** [reply]
Oh, we can probably ping some other macOS people and ask if it normally takes that long for them.

@gaecoli @square-li What do you both reckon of the above? :smile:

**Reply by eugeneihde** [reply, marked as answer]
I now increased the allocated RAM Size for Docker from 2 to 8 GB and now it ran much faster and completed the build.

**Reply by gaecoli** [reply]
![image](https://github.com/getredash/redash/assets/46274612/72504d18-a27d-4fab-a7a2-12eeb784bb5e)

It's ok for me!

**Reply by justinclift** [reply]
Oh yeah, 2GB won't make it happy at all.  Glad you got it figured out. :smile:

**Reply by justinclift** [reply]
@square-li It's probably worth mentioning the above ram size change in the macOS specific developer setup page. :smile:
